File: 2. Two pointer/11. 盛最多水的容器/11. 盛最多水的容器.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int res = 0;
        int l = 0, r = height.size()-1;
        while(l<r){
            int l_value = height[l];
            int r_value = height[r];
            res = max(res, min(l_value, r_value)*(r-l));
            if(l_value < r_value){
                l += 1;
            }
            else{
                r -= 1;
            }
        }
        return res;
    }
};

File: Two pointer/11. 盛最多水的容器/11. 盛最多水的容器.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        res = 0
        while l < r:
            l_value = height[l]
            r_value = height[r]
            res = max(min(l_value, r_value)*(r - l), res)
            if l_value < r_value:
                l += 1
            else:
                r -= 1
        return res